### PR TITLE
Cover cases where 'reverse' field is empty

### PR DESCRIPTION
**Pull request by NatashaL**

### CODE REVIEW

File: styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java [hunk 1]
```diff
@@ -849,8 +849,13 @@ static Backfill entityToBackfill(Entity entity) {
         .nextTrigger(timestampToInstant(entity.getTimestamp(PROPERTY_NEXT_TRIGGER)))
         .schedule(Schedule.parse(entity.getString(PROPERTY_SCHEDULE)))
         .allTriggered(entity.getBoolean(PROPERTY_ALL_TRIGGERED))
-        .halted(entity.getBoolean(PROPERTY_HALTED))
-        .reverse(Optional.ofNullable(entity.getBoolean(PROPERTY_REVERSE)).orElse(false));
+        .halted(entity.getBoolean(PROPERTY_HALTED));
+
+    if (entity.contains(PROPERTY_REVERSE)) {
+      builder.reverse(entity.getBoolean(PROPERTY_REVERSE));
```
Comment (honnix) [inline styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java]: `.reverse(entity.contains(PROPERTY_REVERSE) && entity.getBoolean(PROPERTY_REVERSE))` would this be more fluent or actually harder to read?

File: styx-common/src/test/java/com/spotify/styx/storage/DatastoreStorageTest.java [hunk 2]
```diff
@@ -636,6 +647,37 @@ public void shouldStoreAndReadBackfill(String reverse) throws Exception {
...
+        .set(PROPERTY_START, instantToTimestamp(backfill.start()))
+        .set(PROPERTY_END, instantToTimestamp(backfill.end()))
+        .set(PROPERTY_COMPONENT, backfill.workflowId().componentId())
+        .set(PROPERTY_WORKFLOW, backfill.workflowId().id())
+        .set(PROPERTY_SCHEDULE, backfill.schedule().toString())
+        .set(PROPERTY_NEXT_TRIGGER, instantToTimestamp(backfill.nextTrigger()))
+        .set(PROPERTY_ALL_TRIGGERED, backfill.allTriggered())
+        .set(PROPERTY_HALTED, backfill.halted());
+
+    backfill.description().ifPresent(x -> builder.set(PROPERTY_DESCRIPTION, StringValue
```
Comment (honnix) [inline styx-common/src/test/java/com/spotify/styx/storage/DatastoreStorageTest.java]: Is this to make test case more complete? `description` is empty by definition.

**Comment by honnix** [comment]
LGTM apart from some nitpicking.

File: styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java (hunk 1 above)
Comment (NatashaL) [reply styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java]: I wanted to be consistent with how description was handled. But indeed, personally I find the if-else easier and quicker to understand and using the boolean logic

File: styx-common/src/test/java/com/spotify/styx/storage/DatastoreStorageTest.java (hunk 2 above)
Comment (NatashaL) [reply styx-common/src/test/java/com/spotify/styx/storage/DatastoreStorageTest.java]: No reason really, it was just a quick whip up meant to be a hotfix, but resorted to holding it off a bit. I can remove that for better cleanliness.

File: styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java (hunk 1 above)
Comment (danielnorberg) [reply styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java]: Use `read()`?

```java
.reverse(read(entity, PROPERTY_REVERSE, Boolean.FALSE))
```

Comment (NatashaL) [reply styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java]: `read()` is an object method, can't access it from the static `entityToBackfill()`. I'll end up doing 
```
DatastoreStorage.<Boolean>readOpt(entity, PROPERTY_REVERSE).orElse(Boolean.FALSE)
```
instead

Comment (danielnorberg) [reply styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java]: Feel free to change `entityToBackfill()` to not be static or make `read()` static as well.

File: styx-api-service/src/test/java/com/spotify/styx/api/BackfillResourceTest.java [hunk 3]
```diff
@@ -602,6 +602,31 @@ public void shouldUpdateBackfill() throws Exception {
     assertThat(storage.backfill(BACKFILL_1.id()).get().concurrency(), equalTo(4));
   }
 
+  @Test
+  public void shouldHaltBackfillReversed() throws Exception {
```
Comment (honnix) [inline styx-api-service/src/test/java/com/spotify/styx/api/BackfillResourceTest.java]: Is there anything special when halting a reverse backfill?

Comment (NatashaL) [reply styx-api-service/src/test/java/com/spotify/styx/api/BackfillResourceTest.java]: Halting goes through `retrieveBackfillStatuses` and calculating `instantsInRange`

**Comment by honnix** [comment]
👍

**Pull request merged**